modules/puppet_proxy_legacy/plugin_configuration.rb

```diff
@@ -0,0 +1,109 @@
+                                     else
+                                       :legacy_parser
+                                     end
+
+      force = to_bool(settings[:use_environment_api], nil)
+      settings[:environments_retriever] = if settings[:puppet_version].to_s < '3.2'
+                                            :config_file
+                                          elsif !force.nil? && force
+                                            :api_v2
+                                          elsif !force.nil? && !force
+                                            :config_file
+                                          else
+                                            use_environment_api?(puppet_configuration) ? :api_v2 : :config_file
+                                          end
+
+      settings
+    end
+
+    def load_classes
+      require 'puppet_proxy_common/custom_validators'
+      require 'puppet_proxy_legacy/puppet_config'
+      require 'puppet_proxy_common/errors'
+      require 'puppet_proxy_common/environments_retriever_base'
+      require 'puppet_proxy_legacy/class_scanner_base'
+      require 'puppet_proxy_common/environment'
+      require 'puppet_proxy_common/puppet_class'
+      require 'puppet_proxy_common/api_request'
+
+      require 'puppet'
+      require 'puppet_proxy_legacy/initializer'
+      require 'puppet_proxy_legacy/environments_api_request'
+    end
+
+    def load_dependency_injection_wirings(container_instance, settings)
+      case settings[:environments_retriever]
+        when :api_v2
+          require 'puppet_proxy_legacy/puppet_api_v2_environments_retriever'
+          container_instance.dependency :environment_retriever_impl, (lambda do
+            ::Proxy::PuppetLegacy::PuppetApiV2EnvironmentsRetriever.new(settings[:puppet_url], settings[:puppet_ssl_ca], settings[:puppet_ssl_cert], settings[:puppet_ssl_key])
+          end)
+        else
+          require 'puppet_proxy_legacy/puppet_config_environments_retriever'
+          container_instance.dependency :puppet_configuration,  lambda {Proxy::PuppetLegacy::ConfigReader.new(settings[:puppet_conf])}
+          container_instance.dependency :environment_retriever_impl, lambda {::Proxy::PuppetLegacy::PuppetConfigEnvironmentsRetriever.new(c.get_dependency(:puppet_configuration), settings[:puppet_conf])}
+      end
+
+      container_instance.dependency :puppet_initializer, lambda {Proxy::PuppetLegacy::Initializer.new(settings[:puppet_conf]) }
+
+      case settings[:classes_retriever]
+        when :cached_future_parser
+          require 'puppet_proxy_legacy/class_scanner_eparser'
+          require 'puppet_proxy_legacy/puppet_cache'
+          container_instance.singleton_dependency :class_retriever_impl, (lambda do
+            ::Proxy::PuppetLegacy::PuppetCache.new(c.get_dependency(:environment_retriever_impl), ::Proxy::PuppetLegacy::ClassScannerEParser.new(nil, c.get_dependency(:puppet_initializer)))
```

Comment:
ugh, yeah, this is remnant from when these lambdas had a parameter. I wonder if I should write tests that verify that each of the dependencies can be instantiated; atm I have for the dependencies that have settings as parameters (to validate that parameters are being passed correctly).